### PR TITLE
Fix incorrect gradient in custom_gradient example

### PR DESCRIPTION
**Pull request by ojanik**
Fix incorrect gradient in custom_gradient example

### CODE REVIEW

File: jax/_src/custom_derivatives.py [hunk 1]
```diff
@@ -1150,12 +1150,12 @@ def custom_gradient(fun):
 
   >>> @jax.custom_gradient
   ... def f(x):
-  ...   return x ** 2, lambda g: (g * x,)
+  ...   return x ** 2, lambda g: (g * 2 * x,)
```
Comment (yashk2810) [inline jax/_src/custom_derivatives.py:1153]: Hi! This is on purpose to show that you can return any gradient you want. Can you add a comment instead?
